api/src/main/java/org/apache/iceberg/view/ViewBuilder.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.ViewCatalog;
+
+/**
+ * A builder used to create or replace a SQL {@link View}.
+ *
+ * <p>Call {@link ViewCatalog#buildView} to create a new builder.
+ */
+public interface ViewBuilder {
+  /**
+   * Set the view schema.
+   *
+   * @param schema view schema
+   * @return this for method chaining
+   */
+  ViewBuilder withSchema(Schema schema);
+
+  /**
+   * Set the view query.
+   *
+   * @param query view query
+   * @return this for method chaining
+   */
+  ViewBuilder withQuery(String query);
```

Comment:
If this is supposed to return `SQLViewReperentation`, then it should enforce it somehow?

Reply:
> If this is supposed to return `SQLViewReperentation`, then it should enforce it somehow?

`ViewVersion` returns a list of `ViewRepresentation`. How about adding a field explicitly for `SQLViewReperentation`? In this way, we enforce views should have at least SQL view representation. On disk, we can still lump together and store as a JSON array.

Reply:
Good idea since we always have  at least 1 SQL view representation.

Reply:
If there are multiple SQL view representations for different dialects, I assume the first/default one (specified in buildView) should be set in this new field.

Reply:
Added `ViewVersion.sqlViewRepresentation()`.

Reply:
I think this should be fine for now because it for now we will practically always have sql representation. Down the line, if engines use something use a library like Substrait to serialize common plans then this assumption may not hold.

Reply:
if this is intended for SQL view/query, should this method be called `withSQLQuery`?

Reply:
My original question was that this builder returns a `View`, and in this discussion, you mentioned that `withQuery()` is the method responsible for the fact that it should return `SQLViewRepresentation`. However, I think this method still does not enforce it in this iteration.